### PR TITLE
Adds Beaker acceptance Tests

### PR DESCRIPTION
**Pull request by petems**
Working acceptance tests! :+1:

Blocked until https://github.com/puppetlabs/beaker/pull/737/ is merge, tagged and released so we can switch to beaker master

### CODE REVIEW

**Comment by petems** [comment]
To run: `bundle exec rspec spec/acceptance/chocolatey_provider_spec.rb`

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,12 @@ if ENV.key?('PUPPET_VERSION')
   puppet_version = "= #{ENV['PUPPET_VERSION']}"
 end
 
+gem 'beaker',
+    :git => 'https://github.com/petems/beaker.git',
+    :branch => 'MAINT-non_cygwin_windows_fixes',
+    :require => false
+gem 'beaker-rspec',                                                            :require => false
+
```
Comment (ferventcoder) [inline Gemfile]: Beaker works on Windows now? If not, we'll need to restrict this down to just ruby

File: spec/acceptance/chocolatey_provider_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        }
+      EOS
+
+      apply_manifest(install_chocolatey, :catch_failures => true)
+
+      apply_manifest(choco_notepad_plus, :catch_failures => true)
+      apply_manifest(choco_notepad_plus, :catch_changes  => true)
+    end
+  end
+end
```
Comment (ferventcoder) [inline spec/acceptance/chocolatey_provider_spec.rb]: Sorry but acceptance need to go to top level acceptance folder, not under spec. :)

File: Gemfile (hunk 1 above)
Comment (petems) [inline Gemfile]: Hmm, I didn't think of that actually...

File: spec/acceptance/chocolatey_provider_spec.rb (hunk 2 above)
Comment (petems) [inline spec/acceptance/chocolatey_provider_spec.rb]: Beaker is currently configured to look under the `spec/acceptance/` folder:

`Host file '/Users/petersouter/projects/puppet-chocolatey/spec/acceptance/nodesets/default.yml' does not exist! (ArgumentError)`

File: spec/spec_helper_acceptance.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
+require 'beaker-rspec/spec_helper'
+require 'beaker-rspec/helpers/serverspec'
+require 'serverspec'
+require 'winrm'
+
+GEOTRUST_GLOBAL_CA = <<-EOM
```
Comment (petems) [inline spec/spec_helper_acceptance.rb]: This block can be replaced with the cert helper when https://github.com/puppetlabs/beaker/pull/692 is merged :+1:

File: spec/acceptance/chocolatey_provider_spec.rb (hunk 2 above)
Comment (ferventcoder) [inline spec/acceptance/chocolatey_provider_spec.rb]: For Puppet it works okay under top level acceptance. Plus this pattern breaks my `bundle exec rspec spec`

**Comment by ferventcoder** [comment]
Since this is now in puppetlabs/puppetlabs-chocolatey, I think we can close it over here as that code will flow back down this way. Thanks for the contribution, and sorry it didn't get more of a review than it did.

**Pull request closed without merging**